lib/resty/websocket/client.lua

```diff
@@ -161,6 +168,15 @@ function _M.connect(self, uri, opts)
         return 1
     end
 
+    local custom_headers
+    if headers then
+      local tmp = {}
```

Comment:
Better recycle this Lua table and avoid allocating a new temp table on such a hot code path.

Reply:
@agentzh how would you handle this? Shall we use `lua-tablepool`?

Reply:
@salazar No, the use of the tmp table in this Lua function does not have any Lua API calls that may yield, so lua-tablepool is overkill for this case. Now that you already use a Lua table that can be concatenated directly, the `tmp` table is no longer needed anyway.

Reply:
@agentzh how would you approach that otherwise - for future reference?

Reply:
@salazar Just share a table object in an outer-scope Lua variable.